tests/identify_test.py

```diff
@@ -51,6 +53,18 @@ def test_tags_from_path_symlink(tmpdir):
     assert identify.tags_from_path(x.strpath) == {'symlink'}
 
 
+def test_tags_from_path_socket(tmpdir):
+    cwd = os.getcwd()
```

Comment:
is the chdir necessary? -- should be able to just do:

```python
socket_path = tmpdir.join('socket')
sock = socket.socket(socket.AF_UNIX)
sock.bind(str(socket_path))
# TODO: probably need cleanup code of the socket

assert identify.tags_from_path(socket_path) == {'socket'}

Reply:
My first attempt was like that but I got an error that the path was too long. From some cursory Googling it looks like there is a hardcoded limit that I can't really change, but using a relative path worked

```console
$ pytest --pdb -k socket tests/identify_test.py
======================================================================================================================= test session starts =======================================================================================================================
platform darwin -- Python 3.8.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /Users/mxr/src/go/src/github.com/mxr/identify
plugins: cov-2.11.1
collected 83 items / 82 deselected / 1 selected                                                                                                                                                                                                                   

tests/identify_test.py F
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

tmpdir = local('/private/var/folders/3l/njs4g0nx77v3z1d5085tqtlh0000gp/T/pytest-of-mxr/pytest-24/test_tags_from_path_socket0')

    def test_tags_from_path_socket(tmpdir):
        socket_path = tmpdir.join('socket')
        sock = socket.socket(socket.AF_UNIX)
>       sock.bind(str(socket_path))
E       OSError: AF_UNIX path too long

tests/identify_test.py:59: OSError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /Users/mxr/src/go/src/github.com/mxr/identify/tests/identify_test.py(59)test_tags_from_path_socket()
-> sock.bind(str(socket_path))
(Pdb) socket_path
local('/private/var/folders/3l/njs4g0nx77v3z1d5085tqtlh0000gp/T/pytest-of-mxr/pytest-24/test_tags_from_path_socket0/socket')

```

Reply:
changing directories shouldn't make a difference there -- the socket'll still have the same length (though global iirc if you don't specify an absolute path)

if you use `TemporaryDirectory(dir='/tmp')` does it improve the problem? -- macos is kinda bad about having short tempdirs

Reply:
> changing directories shouldn't make a difference there -- the socket'll still have the same length (though global iirc if you don't specify an absolute path)

Maybe what's happening is socket creation succeeds (evidenced in the test) but subsequent use of the socket will fail in other ways?

> if you use `TemporaryDirectory(dir='/tmp')` does it improve the problem? -- macos is kinda bad about having short tempdirs

Sure I'll try that instead